### PR TITLE
 Issue #47 and Issue #48

### PR DESCRIPTION
**Pull request by swarwick**

### CODE REVIEW

**Comment by swarwick** [comment]
Are you guys reviewing this?

File: mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/ChannelControllerRouter.java [hunk 1]
```diff
@@ -131,14 +133,25 @@ protected void routeMBPEntry(int securityId, MdpMessage mdpMessage, MdpGroupEntr
...
+            InstrumentController instrumentController = instrumentManager.getInstrumentController(securityId);
+            if (instrumentController != null) {
+                instrumentController.handleIncrementalComplete(mdpMessage, msgSeqNum);
+            }
+        }
+    }
+    
     private void handleIncrementalMessage(MdpMessage mdpMessage, MdpGroup mdpGroup, MdpGroupEntry mdpGroupEntry, long msgSeqNum){
         if (isIncrementalMessageSupported(mdpMessage)) {
+            Set<Integer> securityIds = new HashSet<>();
```
Comment (kolybelkin) [inline mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/ChannelControllerRouter.java]: we can't create a new object every time we receive and handle a message it will lead to performance degradation and pauses

**Comment by iamolever** [comment]
> Are you guys reviewing this?

Hi Steven, yes, we are reviewing.

File: mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/OffHeapSnapshotCycleHandler.java [hunk 2]
```diff
@@ -123,7 +123,7 @@ private void clearArray(LongArray array) {
     }
 
     private class LongArray {
-        private BytesStore bytes;
+        private BytesStore<?, ?> bytes;
```
Comment (kolybelkin) [inline mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/OffHeapSnapshotCycleHandler.java]: Why did you add typification here?

File: mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/ChannelControllerRouter.java (hunk 1 above)
Comment (swarwick) [reply mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/ChannelControllerRouter.java]: true, could create at top level and clear

File: mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/OffHeapSnapshotCycleHandler.java (hunk 2 above)
Comment (swarwick) [reply mbp-with-mbo/src/main/java/com/epam/cme/mdp3/control/OffHeapSnapshotCycleHandler.java]: Because the interface has them net.openhft.chronicle.bytes.BytesStore<?, ?>

**Comment by iamolever** [comment]
Let's reuse hashset and after that PR looks OK for me

ср, 3 окт. 2018 г. в 20:56, Steven Warwick <notifications@github.com>:

**Comment by swarwick** [comment]
Checked in change for reuse hashset

**Comment by swarwick** [comment]
After the pull requests are complete who does the actual release builds?  Can we do another one since its an interface and version bump?

**Comment by iamolever** [comment]
Hi @kolybelkin, do you agree with the changes now?

**Comment by kolybelkin** [comment]
I would use agrona collections here or similar Zero-GC libraries

**Comment by kolybelkin** [comment]
Looks goods to me now. We need to publish the new version on releases tab with the release notes after merging it. @iamolever ?
